internal/releasesources/release_sources.go

```diff
@@ -0,0 +1,413 @@
+		if err != nil {
+			return err
+		}
+
+		if *majorMinorVersion == openshiftMajorMinorVersion && release.Channel == "candidate" {
+			releasesToDelete = append(releasesToDelete, release)
+		}
+	}
+	// Nothing to delete
+	if len(releasesToDelete) == 0 {
+		return nil
+	}
+
+	return db.Where("TRUE").Delete(&releasesToDelete).Error
+}
+
+func setSupportLevel(db *gorm.DB, version Version) error {
+	var (
+		releases []common.OpenshiftVersion
+	)
+	err := db.Model(&common.OpenshiftVersion{}).
+		Find(&releases).Error
+	if err != nil {
+		return err
+	}
+
+	for _, release := range releases {
+		releaseCopy := release
+		majorMinorVersion, err := common.GetMajorMinorVersion(*release.OpenshiftVersion.DisplayName)
+		if err != nil {
+			return err
+		}
+
+		if *majorMinorVersion == version.Name {
+			releaseCopy.SupportLevel = &version.Type
+			err = db.Save(&releaseCopy).Error
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func setDefaultRelease(db *gorm.DB) error {
+	var (
+		defaultCandidatesStableReleases []common.OpenshiftVersion
+		err                             error
+		lessThan                        bool
+	)
+	err = db.Model(&common.OpenshiftVersion{}).
+		Where("channel = 'stable' AND ? = ANY(cpu_architectures)", defaultCPUArchitecture).
+		Find(&defaultCandidatesStableReleases).Error
+	if err != nil {
+		return err
+	}
+
+	if len(defaultCandidatesStableReleases) == 0 {
+		return fmt.Errorf("no stable releases with cpu architecture %s found", defaultCPUArchitecture)
+	}
+
+	latestReleaseVersion := defaultCandidatesStableReleases[0].DisplayName
+	latestStableRelease := defaultCandidatesStableReleases[0]
+
+	for _, release := range defaultCandidatesStableReleases {
+		lessThan, err = common.BaseVersionLessThan(*release.DisplayName, *latestReleaseVersion)
+		if err != nil {
+			return err
+		}
+
+		if lessThan {
+			latestReleaseVersion = release.DisplayName
+			latestStableRelease = release
+		}
+	}
+
+	// Changing to new default
+	err = db.Model(&common.OpenshiftVersion{}).
+		Where("\"default\" = ?", true).
+		Update("\"default\"", false).Error
+	if err != nil {
+		return err
+	}
+
+	latestStableRelease.Default = true
+
+	return db.Save(&latestStableRelease).Error
+
+}
+
+func syncOpenshiftReleasesMetadata(db *gorm.DB, client OpenShiftSupportLevelAPIClient) error {
+	mapAPISupportLevelToOurSupportLevel := map[string]string{
+		"End of life":         "end of life",
+		"Maintenance Support": "maintenance",
+		"Full Support":        "production",
+	}
+
+	supportLevelGraph, err := client.getSupportLevels(openshiftMajorVersion)
+	if err != nil {
+		return err
+	}
+
+	for _, Data := range supportLevelGraph.Data {
+		for _, version := range Data.Versions {
+			if convertion, ok := mapAPISupportLevelToOurSupportLevel[version.Type]; ok {
+				version.Type = convertion
+			}
+
+			err = deletePreReleaseVersions(db, version.Name)
+			if err != nil {
+				return err
+			}
+
+			err = setSupportLevel(db, version)
```

Comment:
Can we try to unify the insert to contain this data too, without adding complexity in the code?
This will add N inserts + one update on few rows later. Not a big deal but not necessary